orchestrator/websocket/__init__.py

```diff
@@ -118,7 +130,7 @@ def send_process_data_to_websocket(
         logger.debug("Broadcast process data directly to websocket_manager", process_id=str(process_id))
 
         anyio.run(websocket_manager.broadcast_data, [WS_CHANNELS.ALL_PROCESSES], data)
-        sync_broadcast_invalidate_cache("processes", str(process_id))
+        sync_broadcast_invalidate_cache("processes", f"process:{process_id}")
```

Comment:
As far as I remember we were using arrays as cache keys, so something like

```
["processes", <uuid>]
["subscriptions", <uuid>]
```

Did that format change to colon-separated strings now? 

@DutchBen to verify from frontend perspective

Reply:
An array as cache key sounds like a bad design choice. Are you sure it's not for sending multiple cache keys in a single send?

Reply:
It's not meant for sending multiple keys, just an alternative to a delimited string. What's particularly bad about it?

I don't have a preference for one or the other, I'm fine withing a string delimited by `:` or `.` or something else. We'd just have to do this in accordance with the UI.

Reply:
I was interpreting the message ["processes", <uuid>] as one message and the uuid as a processes uuid. I understand from @tjeerddie that it is in fact 2 messages 'processes' and a uuid that is not necessarily a process UUID. I can go either way with a slight preference for one message per message.

Reply:
Cache keys need to be checked for equality and checking arrays for equality in JS can easily lead to ambiguities, since they are not values like strings are.

Or, if you stringify them to JSON, is `["process","123"] == ["process", "123"]`?

Reply:
@DutchBen showed me how we use the caching on the frontend.
which is a object `{ type: 'processes' }` for list and `{ type: 'processes', id: "uuid" }` for a process detail.
I suggest we send it exactly like that, so its consistent on both frontend and backend, removing confusion on how one would add their own cache invalidation with both projects.

and do we keep the list or send one object per websocket message?

Reply:
So just to make sure we're on the same page, the messages as a whole currently look like this;

```json
{"name": "invalidateCache", "value": ["processes", "(the id here)"]}

{"name": "invalidateCache", "value": ["engineStatus"]}
```

And we'd want to change them to look like below?

```json
{"name": "invalidateCache", "value": {"type": "processes", "id": "(the id here)"}}

{"name": "invalidateCache", "value": {"type": "engineStatus"}}
```

If so: it's more explicit, I think it's great. 

> and do we keep the list or send one object per websocket message?

I think one object is fine.

Reply:
> Cache keys need to be checked for equality and checking arrays for equality in JS can easily lead to ambiguities, since they are not values like strings are.

Fair point, I didn't know this was ambiguous in JS. I agree to change to an object as above.